exporter/awscloudwatchlogsexporter/config.go

```diff
@@ -42,6 +44,30 @@ type Config struct {
 	// e.g. logs.us-east-1.amazonaws.com
 	// Optional.
 	Endpoint string `mapstructure:"endpoint"`
+
+	// QueueSettings is a subset of exporterhelper.QueueSettings,
+	// because only QueueSize is user-settable due to how AWS CloudWatch API works
+	QueueSettings struct {
+		// QueueSize set the lenght of the sending queue
+		QueueSize int `mapstructure:"queue_size"`
+	} `mapstructure:"sending_queue"`
+}
+
+// Validate config
+func (config *Config) Validate() error {
+	if config.QueueSettings.QueueSize < 1 {
+		return errors.New("sending_queue.queue_size must be 1 or greater")
+	}
+	return nil
+}
+
+func (config *Config) enforcedQueueSettings() exporterhelper.QueueSettings {
```

Comment:
@bogdandrutu @mxiamxia I think this function should make the intention a bit clearer, thoughts?